contracts/rbn-staking/FeeCustody.sol

```diff
@@ -0,0 +1,425 @@
+    address _feeDistributor,
+    address _protocolRevenueRecipient,
+    address _admin,
+    address _keeper
+  ) {
+    require(_feeDistributor != address(0), "!_feeDistributor");
+    require(
+      _protocolRevenueRecipient != address(0),
+      "!_protocolRevenueRecipient"
+    );
+    require(_admin != address(0), "!_admin");
+    require(_keeper != address(0), "!_keeper");
+
+    pctAllocationForRBNLockers = _pctAllocationForRBNLockers;
+    feeDistributor = IFeeDistributor(_feeDistributor);
+    protocolRevenueRecipient = _protocolRevenueRecipient;
+    keeper = _keeper;
+  }
+
+  /**
+   * @notice
+   * Swaps RBN locker allocation of protocol revenu to distributionToken,
+   * sends the rest to the multisig
+   * @dev Can be called by keeper
+   * @param _minAmountOut min amount out for every asset type swap.
+   * will need to be in order of assets in assets[] array. should be
+   * fine if we keep track.
+   * @return toDistribute amount of distributionToken distributed to fee distributor
+   */
+  function distributeProtocolRevenue(uint256[] calldata _minAmountOut)
+    external
+    returns (uint256 toDistribute)
+  {
+    require(msg.sender == keeper, "!keeper");
+
+    if (address(distributionToken) == WETH) {
+      IWETH(address(distributionToken)).deposit{value: address(this).balance}();
+    }
+
+    for (uint256 i; i < lastAssetIdx; i++) {
+      IERC20 asset = IERC20(assets[i]);
+      uint256 assetBalance = asset.balanceOf(address(this));
+
+      if (assetBalance == 0) {
+        continue;
+      }
+
+      uint256 multiSigRevenue = assetBalance
+        .mul(TOTAL_PCT.sub(pctAllocationForRBNLockers))
+        .div(TOTAL_PCT);
+
+      // If we are holding the distributionToken itself,
+      // do not swap
+      if (address(asset) != address(distributionToken)) {
+        // Calculate RBN allocation amount to swap for distributionToken
+        uint256 amountIn = assetBalance.sub(multiSigRevenue);
+        _swap(address(asset), amountIn, _minAmountOut[i]);
+      }
+
+      // Transfer multisig allocation of protocol revenue to multisig
+      asset.safeTransfer(protocolRevenueRecipient, multiSigRevenue);
+    }
+
+    toDistribute = distributionToken.balanceOf(address(this));
+    distributionToken.safeApprove(address(feeDistributor), 0);
+    distributionToken.safeApprove(address(feeDistributor), toDistribute);
+
+    // Tranfer RBN locker allocation of protocol revenue to fee distributor
+    feeDistributor.burn(address(distributionToken), toDistribute);
+  }
+
+  /**
+   * @notice
+   * Amount of _asset allocated to RBN lockers from current balance
+   * @return amount allocated to RBN lockers
+   */
+  function claimableByRBNLockersOfAsset(address _asset)
+    external
+    view
+    returns (uint256)
+  {
+    uint256 allocPCT = pctAllocationForRBNLockers;
+    uint256 balance = _asset == address(0)
+      ? address(this).balance
+      : IERC20(_asset).balanceOf(address(this));
```

Comment:
How would you handle a combination of WETH + ETH here? If the FeeCustody contract holds both WETH + ETH, we will only count the ETH on the contract.

Reply:
you would have to get claimable separately - once for eth (`claimableByRBNLockersOfAsset(address(0))`) and once for weth(`claimableByRBNLockersOfAsset(address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2))`